engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -2081,6 +2081,10 @@ public void updateRecoveryCheckpoints(PTOperator operator, UpdateCheckpointsCont
     if (checkpointGroup.size() > 1) {
       for (OperatorMeta om : checkpointGroup) {
         Collection<PTOperator> operators = plan.getAllOperators(om);
```

Comment:
Yes, the documentation isn't correct though, this applies not just to a shuffle, but even to a single downstream operator (that was the change I referred to).